diff --git a/node_modules/react-day-picker/dist/index.js b/node_modules/react-day-picker/dist/index.js
index 9295551c82e36122d650118804595e803a5538d3..02e0797171e285d8520377c8e967a57a5522778a 100644
--- a/node_modules/react-day-picker/dist/index.js
+++ b/node_modules/react-day-picker/dist/index.js
@@ -193,7 +193,7 @@ function useInput(
   Required<Pick<UseInputOptions, "locale" | "required">> {
   const [month, setMonth] = React.useState(
     () => options.defaultMonth ?? new Date()
   );
-  const [selected, setSelected] = React.useState(options.defaultSelected);
+  const [selected, setSelected] = React.useState(options.defaultSelected);  
   const [inputValue, setInputValue] = React.useState("");
   React.useEffect(() => {
     if (!options.defaultSelected) {
@@ -201,15 +201,15 @@
     }
     setInputValue(
       formatValue(options.defaultSelected, options.format, options.locale)
     );
-  }, [options.defaultSelected]);
+  }, [options.defaultSelected, options.format, options.locale]);
   const [lastValidValue, setLastValidValue] = React.useState(
     options.defaultSelected
   );
   const dayPickerRef = React.useRef(null);
-  const handleDayClick: DayClickEventHandler = (day, modifiers) => {
+  const handleDayClick = (day, modifiers) => {
     if (modifiers.selected) {
       if (options.required) {
         return;
@@ -231,14 +231,14 @@ function useInput(
     }
     dayPickerRef.current?.focus();
   };
-  const handleMonthChange: MonthChangeEventHandler = (month2) => {
+  const handleMonthChange = (month2) => {
     setMonth(month2);
   };
   const ISO_DATE_FORMAT = "yyyy-MM-dd";
   const parseValue = (value) => {
     if (!value) {
       return void 0;
     }
     if (options.format) {
       try {
@@ -259,7 +259,7 @@ function useInput(
     }
     return date;
   };
-  const formatValue = (date) => {
+  const formatValue = (date, format, locale) => {
     if (!date) {
       return "";
     }
@@ -270,7 +270,7 @@ function useInput(
     return format(date, options.format, { locale: options.locale });
   };
   const handleBlur: React.FocusEventHandler = (e) => {
-    const value = e.target.value;
+    const value = e.currentTarget.value;
     const date = parseValue(value);
     if (isValidDate(date)) {
       setLastValidValue(date);
@@ -279,15 +279,14 @@ function useInput(
     } else {
       setSelected(lastValidValue);
       setInputValue(
-        formatValue(
-          lastValidValue,
-          options.format,
-          options.locale
-        )
+        formatValue(lastValidValue, options.format, options.locale)
       );
     }
     options.onBlur?.(e);
   };
-  const handleChange: React.ChangeEventHandler = (e) => {
-    const value = e.target.value;
+  const handleChange = (e) => {
+    const value = e.currentTarget.value;
     setInputValue(value);
     const date = parseValue(value);
     if (!isValidDate(date)) {
@@ -300,7 +299,7 @@ function useInput(
   };
   const handleReset = () => {
     setSelected(void 0);
-    setInputValue("");
+    setInputValue(formatValue(void 0, options.format, options.locale));
   };
   return {
     /** The props for the DayPicker component. */
diff --git a/node_modules/react-day-picker/dist/index.mjs b/node_modules/react-day-picker/dist/index.mjs
index 455af2b89643d9c824c3a2f3493d6d538e12d4d8..e32d667c30932de071d2b7c4d5d9086c8770c18d 100644
--- a/node_modules/react-day-picker/dist/index.mjs
+++ b/node_modules/react-day-picker/dist/index.mjs
@@ -194,7 +194,7 @@ function useInput(
   Required<Pick<UseInputOptions, "locale" | "required">> {
   const [month, setMonth] = React.useState(
     () => options.defaultMonth ?? new Date()
   );
-  const [selected, setSelected] = React.useState(options.defaultSelected);
+  const [selected, setSelected] = React.useState(options.defaultSelected);  
   const [inputValue, setInputValue] = React.useState("");
   React.useEffect(() => {
     if (!options.defaultSelected) {
@@ -202,15 +202,15 @@
     }
     setInputValue(
       formatValue(options.defaultSelected, options.format, options.locale)
     );
-  }, [options.defaultSelected]);
+  }, [options.defaultSelected, options.format, options.locale]);
   const [lastValidValue, setLastValidValue] = React.useState(
     options.defaultSelected
   );
   const dayPickerRef = React.useRef(null);
-  const handleDayClick: DayClickEventHandler = (day, modifiers) => {
+  const handleDayClick = (day, modifiers) => {
     if (modifiers.selected) {
       if (options.required) {
         return;
@@ -232,14 +232,14 @@ function useInput(
     }
     dayPickerRef.current?.focus();
   };
-  const handleMonthChange: MonthChangeEventHandler = (month2) => {
+  const handleMonthChange = (month2) => {
     setMonth(month2);
   };
   const ISO_DATE_FORMAT = "yyyy-MM-dd";
   const parseValue = (value) => {
     if (!value) {
       return void 0;
     }
     if (options.format) {
       try {
@@ -260,7 +260,7 @@ function useInput(
     }
     return date;
   };
-  const formatValue = (date) => {
+  const formatValue = (date, format, locale) => {
     if (!date) {
       return "";
     }
@@ -271,7 +271,7 @@ function useInput(
     return format(date, options.format, { locale: options.locale });
   };
   const handleBlur: React.FocusEventHandler = (e) => {
-    const value = e.target.value;
+    const value = e.currentTarget.value;
     const date = parseValue(value);
     if (isValidDate(date)) {
       setLastValidValue(date);
@@ -280,15 +280,14 @@ function useInput(
     } else {
       setSelected(lastValidValue);
       setInputValue(
-        formatValue(
-          lastValidValue,
-          options.format,
-          options.locale
-        )
+        formatValue(lastValidValue, options.format, options.locale)
       );
     }
     options.onBlur?.(e);
   };
-  const handleChange: React.ChangeEventHandler = (e) => {
-    const value = e.target.value;
+  const handleChange = (e) => {
+    const value = e.currentTarget.value;
     setInputValue(value);
     const date = parseValue(value);
     if (!isValidDate(date)) {
@@ -301,7 +300,7 @@ function useInput(
   };
   const handleReset = () => {
     setSelected(void 0);
-    setInputValue("");
+    setInputValue(formatValue(void 0, options.format, options.locale));
   };
   return {
     /** The props for the DayPicker component. */
diff --git a/node_modules/react-day-picker/package.json b/node_modules/react-day-picker/package.json
index 515865882e36780c12852233a758797f742c8d23..5f01d41875c7423e2185d2639b031c6a256df901 100644
--- a/node_modules/react-day-picker/package.json
+++ b/node_modules/react-day-picker/package.json
@@ -62,7 +62,7 @@
     "eslint": "8.57.0",
     "eslint-plugin-promise": "6.1.1",
     "eslint-plugin-react": "7.34.1",
     "eslint-plugin-react-hooks": "4.6.0",
-    "react": "18.2.0",
-    "react-dom": "18.2.0",
+    "react": "19.0.0-rc-f994737d14-20240522",
+    "react-dom": "19.0.0-rc-f994737d14-20240522",
     "react-test-renderer": "18.2.0",
     "rimraf": "5.0.5",
     "ts-node": "10.9.2",
@@ -75,7 +75,7 @@
   "peerDependencies": {
     "date-fns": "^2.30.0 || ^3.0.0",
     "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
-    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0"
+    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0 || ^19.0.0"
   },
   "dependencies": {
     "@babel/runtime": "7.24.1"
diff --git a/node_modules/react-day-picker/dist/hooks/useInput.d.ts b/node_modules/react-day-picker/dist/hooks/useInput.d.ts
index e56b825316682622830f28e2ad3b8110b1016866..c0514fe96b1b59c7f66a7b6703901b5a5b50cd23 100644
--- a/node_modules/react-day-picker/dist/hooks/useInput.d.ts
+++ b/node_modules/react-day-picker/dist/hooks/useInput.d.ts
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { DayPickerProps } from '../types';
 import { DayClickEventHandler, MonthChangeEventHandler } from '../handlers';
-import { DayPicker } from '../components/DayPicker';
+import type { DayPicker } from '../components/DayPicker';
 /** The props for the {@link useInput} hook. */
 export interface UseInputOptions extends Pick<DayPickerProps, 'classNames' | 'styles' | 'components'> {
     /**
diff --git a/node_modules/react-day-picker/dist/components/DayPicker/DayPicker.d.ts b/node_modules/react-day-picker/dist/components/DayPicker/DayPicker.d.ts
index 72545d1d64379a7bb24f331980a373bde1c712ba..a069dfb077a83d722e032ca23d8c114f08f029a1 100644
--- a/node_modules/react-day-picker/dist/components/DayPicker/DayPicker.d.ts
+++ b/node_modules/react-day-picker/dist/components/DayPicker/DayPicker.d.ts
@@ -1,6 +1,5 @@
 import * as React from 'react';
 import { DayPickerProps } from '../../types';
-import { DayPickerProvider } from '../DayPickerProvider';
 /**
  * DayPicker is the main component for rendering a calendar.
  *
@@ -10,6 +9,4 @@
  *
  * @see http://daypicker.dev
  */
-export declare function DayPicker(props: DayPickerProps): JSX.Element;
-export declare namespace DayPicker {
-    var displayName: string;
-}
+export declare const DayPicker: React.ForwardRefExoticComponent<DayPickerProps & React.RefAttributes<HTMLDivElement>>;
\ No newline at end of file
diff --git a/node_modules/react-day-picker/dist/components/DayPicker/index.d.ts b/node_modules/react-day-picker/dist/components/DayPicker/index.d.ts
index 759f237f00dd6657a8a1de2a048a4c3f56b001d9..b850d993132bb82f505342a781b2ca9334bb32f6 100644
--- a/node_modules/react-day-picker/dist/components/DayPicker/index.d.ts
+++ b/node_modules/react-day-picker/dist/components/DayPicker/index.d.ts
@@ -1 +1 @@
-export * from './DayPicker';
\ No newline at end of file
+export { DayPicker } from './DayPicker';
\ No newline at end of file
diff --git a/node_modules/react-day-picker/dist/components/DayPickerProvider/DayPickerProvider.d.ts b/node_modules/react-day-picker/dist/components/DayPickerProvider/DayPickerProvider.d.ts
index 6a5a2283e39b752259160d5b5d1e2333dfd5a9d8..6e42b23a9d9e6022e39130761e3d3ba0d7d74f26 100644
--- a/node_modules/react-day-picker/dist/components/DayPickerProvider/DayPickerProvider.d
+++ b/node_modules/react-day-picker/dist/components/DayPickerProvider/DayPickerProvider.d
@@ -1,13 +1,11 @@
 import * as React from 'react';
 import { DayPickerProps } from '../../types';
-import { DayPickerContext, DayPickerContextValue } from './DayPickerContext';
+import { DayPickerContextValue } from './DayPickerContext';
 export type DayPickerProviderProps = {
     /** The props passed to the DayPicker component. */
     initialProps: DayPickerProps;
     children: React.ReactNode;
 };
 /**
- * The provider for the {@link DayPickerContext}, assigning the defaults from the
- * initial props.
+ * The provider for the DayPicker context, assigning the defaults from the initial props.
  */
 export declare function DayPickerProvider(props: DayPickerProviderProps): JSX.Element;
 export declare namespace DayPickerProvider {
     var displayName: string;
 }
-export type { DayPickerContext, DayPickerContextValue };
+export type { DayPickerContextValue };